File: packages/expo-manifests/ios/EXManifests/EXManifestsBaseLegacyManifest.m
//  Copyright © 2021 650 Industries. All rights reserved.

#import <EXManifests/EXManifestsBaseLegacyManifest.h>
#import <EXJSONUtils/NSDictionary+EXJSONUtils.h>

@implementation EXManifestsBaseLegacyManifest

- (nullable NSDictionary *)expoClientConfigRootObject {
  return self.rawManifestJSON;
}

- (nullable NSDictionary *)expoGoConfigRootObject {
  return self.rawManifestJSON;
}

- (NSString *)stableLegacyId {
  NSString *originalFullName = [self.rawManifestJSON nullableStringForKey:@"originalFullName"];
  if (originalFullName) {
    return originalFullName;
  } else {
    return [self legacyId];
  }
}

- (NSString *)scopeKey {
  NSString *scopeKey = [self.rawManifestJSON nullableStringForKey:@"scopeKey"];
  if (scopeKey) {
    return scopeKey;
  } else {
    return [self stableLegacyId];
  }
}

- (nullable NSString *)easProjectId {
  return [self.rawManifestJSON nullableStringForKey:@"projectId"];
}

- (NSString *)bundleUrl {
  return [self.rawManifestJSON stringForKey:@"bundleUrl"];
}

- (nullable NSString *)sdkVersion {
  return [self.rawManifestJSON nullableStringForKey:@"sdkVersion"];
}

- (nullable NSArray *)assets {
  return [self.rawManifestJSON nullableArrayForKey:@"assets"];
}

@end

File: packages/expo-manifests/ios/EXManifests/EXManifestsBaseLegacyManifest.swift
//  Copyright © 2021 650 Industries. All rights reserved.

import Foundation

@objc public class EXManifestBaseLegacyManifest : EXManifestsManifest {
  override func expoClientConfigRootObject() -> Dictionary<String, Any>? {
    return self.rawManifestJSON()
  }

  override func expoGoConfigRootObject() -> Dictionary<String, Any>? {
    return self.rawManifestJSON()
  }

  @objc public override func stableLegacyId() -> String {
    return self.rawManifestJSON().optionalValue(forKey: "originalFullName") ?? self.legacyId()
  }

  @objc public override func scopeKey() -> String {
    return self.rawManifestJSON().optionalValue(forKey: "scopeKey") ?? self.stableLegacyId()
  }

  @objc public override func easProjectId() -> String? {
    return self.rawManifestJSON().optionalValue(forKey: "projectId")
  }

  @objc public override func bundleUrl() -> String {
    return self.rawManifestJSON().requiredValue(forKey: "bundleUrl")
  }

  @objc public override func sdkVersion() -> String? {
    return self.rawManifestJSON().optionalValue(forKey: "sdkVersion")
  }

  @objc public func assets() -> [Any]? {
    return self.rawManifestJSON().optionalValue(forKey: "assets")
  }
}
